File: DSA/Priyanshi/Day_1_Question_1.java
import java.util.*;

public class Day_1_Question_1 
{
    public static int countWays(int[] candy, int day, int month) {
        int count = 0;
        for (int i = 0; i <= candy.length - month; i++) {
            int sum = 0;
            for (int j = i; j < i + month; j++) {
                sum += candy[j];
            }
            if (sum == day) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        int n=0, day=0, month=0;
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the size of array: ");
        n = sc.nextInt();

        int candy[] = new int[n];

        System.out.println("Enter the integers representing the chocolate squares: ");
        for(int i=0;i<n;i++)
        {
            candy[i] = sc.nextInt();
        }

        System.out.println("Enter the day: ");
        day = sc.nextInt();
        System.out.println("Enter the month: ");
        month = sc.nextInt();

        System.out.println("Output: " + countWays(candy, day, month));
    }
}

File: DSA/Arshdeep Kaur/Day1_Question1.cpp
#include <iostream>
using namespace std;

int main() {
    int candy[10], n, day, month, count = 0;
    cout << "Enter the number of chocolate squares: ";
    cin >> n;
    cout << "Enter the chocolate squares: ";
    for (int i=0; i<n; i++) {
        cin >> candy[i];
    }

    cout << "Enter the day: ";
    cin >> day;
    cout << "Enter the month: ";
    cin >> month;

    for (int i = 0; i <= n - month; ++i) {
        int sum = 0;
        for (int j = 0; j < month; ++j) {
            sum += candy[i + j];
        }
        if (sum == day) {
            count++;
        }
    }

    cout << count << endl;
}
